controllers/storagecluster/reconcile.go

```diff
@@ -723,3 +734,51 @@ func validateOverprovisionControlSpec(sc *ocsv1.StorageCluster, reqLogger logr.L
 	}
 	return nil
 }
+
+func validateCustomStorageClassNames(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	scMap := make(map[string]bool)
+	duplicateNames := []string{}
+	if sc.Spec.ManagedResources.CephBlockPools.StorageClassName != "" {
+		if _, has := scMap[sc.Spec.ManagedResources.CephBlockPools.StorageClassName]; has {
+			duplicateNames = append(duplicateNames, fmt.Sprintf("Duplicate storage class name provided for CephBlockPools: %s", sc.Spec.ManagedResources.CephBlockPools.StorageClassName))
```

Comment:
```suggestion
			duplicateNames = append(duplicateNames, "CephBlockPools"))
```
Just store resources with duplicate storageclass names. Error message is already there in the end, so we can remove duplicates.

Reply:
Ahh ok, updated it now. Sorry for the trouble 😓